### PR TITLE
Ignore COMException which pollutes the debugging log.

### LINKED ISSUE #1580: WebSocket module throws an exception every 5 seconds when you're not using the react-devtools
[This line](https://github.com/Microsoft/react-native-windows/blob/fa85c5778f52d7ed8aae26e37dca97e611b8057a/ReactWindows/ReactNative/Modules/WebSocket/WebSocketModule.cs#L145) in the WebSocket module throws an exception once every 5 seconds.
 
The exception is:

```
WinRT information: A connection with the server could not be established
 
'System.Runtime.InteropServices.COMException' in System.Private.CoreLib.ni.dll
```

This contributes a lot of noise to Visual Studio's debug output window because it gets logged once every five seconds.

@rozele said:

> We can and should definitely do something about it. I forgot about the issue because I have a local copy of the JS module that connects to the react-devtools and I made it a no-op.

### PR DESCRIPTION
**Pull request by leoek**
This fixes #1580 

I am not sure what the correct base branch is, feel free to name me another one if 0.57-stable was incorrect.

### CODE REVIEW

**Comment by leoek** [comment]
Hmm probably its better to merge matchwerk:leoek/master into master

**Comment by rozele** [comment]
Thanks @leoek, I think we may instead want to change the behavior in JavaScript which is continuously trying to connect to the React Debugger. The COM exception is actually a valid exception that the connection failed. We wouldn’t want to hide this exception.

**Comment by leoek** [comment]
@rozele Yes that would be better. Is there progress on this? This Exception is really annoying.

**Comment by acoates-ms** [comment]
The retry of the websocket connection is core RN code, nothing specific to react-native-windows.  You could try pushing an option to facebook to enable it to be disabled in the JS.  (see https://github.com/facebook/react-native/blob/master/Libraries/Core/Devtools/setupDevtools.js).  You can also configure metro to alias setupDevtools to point to an empty js file to prevent the devtools from loading at all.

**Comment by leoek** [comment]
@acoates-ms I actually don't want to disable and enable the retry. The linked source looks fine for me.

I changed the catch clause to be more specific about the errors which are ignored which works pretty fine for me.

File: ReactWindows/ReactNative/Modules/WebSocket/WebSocketModule.cs [hunk 1]
```diff
@@ -152,6 +152,21 @@ private async void InitializeInBackground(int id, string url, MessageWebSocket w
                     { "id", id },
                 });
             }
+            catch (System.Runtime.InteropServices.COMException ex)
+            {
+#if DEBUG
+                if (url.StartsWith("ws://localhost"))
```
Comment (licanhua) [inline ReactWindows/ReactNative/Modules/WebSocket/WebSocketModule.cs]: It lose the first the exception too. but from the bug, it's better to only avoid it happy every 5 seconds.
I suggest you keep the first exception if possible

Comment (leoek) [reply ReactWindows/ReactNative/Modules/WebSocket/WebSocketModule.cs:158]: It doesn't seem necessary to introduce additional state here.

**Comment by asklar** [comment]
closing stale PR, please reopen if this is still needed/applicable.

**Pull request closed without merging**